### ISSUE TITLE
Sybil Attacker Report

### ISSUE THREAD

**Issue by GreyPK** [hop-protocol/hop-airdrop#98]
# Related Addresses

29 related addresses from 1 transaction

```
0x1d9237940b102960aa4655d25f86626b9a8e743a
0x250613f89a1543b1f97be32dc582bd3296941803
0x2a11d404dabbdf799bb6e0b319327f93765b5216
0x2c2f46041f90062636d0edaa0b345e6f71b42933
0x2fdaa34af555ee71ad5b87d4776e4c3e9b35368f
0x3dd52c3385c211bc5a99578d0bf898814b1e48ca
0x55ee4f3769c870d816f2f98bf7258aa4eed8d8eb
0x5afffee90d48a6cd471adefb9cf45cfc3112e538
0x638be8bdd0ee9f614199835bd688b911a8c51793
0x6bfddb07ec564ef6b9b1f4714e3dc6390eb9faef
0x6c66677f31093b465f14cf1455e820e9a86501d3
0x6d45810ed704fcb7d51b60783f493e63604eb717
0x7ac1a4f9802666bfdc3ae72597e8eeee1f77b62d
0x7c9e6c886abb52d24b50f7e759db3af1b6459042
0x9958f1a31c70a97302370364fe701a420cfe5829
0x9fac8fa825c16af889f0283c9bb63ba0ea437c37
0xa51235b7d1087da2c6c5fbf022f48fc3458da11e
0xa84174465090fb5dea838c45c675eaab6cc54a88
0xab421b2e9e0eac0c0f551a6c3b088f37c19534b2
0xabcfb5692b3517449ffb192e7f7516f1bb52975e
0xae8d883df0bbb0ddea5915001eadc4c67b146d6a
0xba4ac6587413c5980ab699cf6d5dcb4562b7d6c8
0xc73765fc508528ba0b6ef13c93e53b7fc579cd81
0xd3ba5e075b93e03b2fb0e2dc72138a37ad9efe78
0xd9a224313a26560bc4a737012cc68c365ad227c4
0xde5a9aa92c8e80a06e6cd8affa11c7695df8f337
0xe3d13050bf335b9c69a9e9e77736351f73fc0e8f
0xece85819cbe493b488ec890c81b9583f9f1c660d
0xf14ca4708db963eb5d4d78fce961c719e5703e5e
```

# Reasoning

1) Address 0x4b1f934c95950f1e9a95ec986680ad3fded1a695 used Disperse.app to send MATIC to these addresses in 1 transaction. 29 of 30 recipients addresses of that thx send is eligible for the airdrop. This transaction: https://polygonscan.com/tx/0xfc99702bde95d1c44bb8ac87178f3b8ec292fd51bd28a1425071cef3dcbc9016
2) All 29 wallets recived 507-511 DAI from HOP with bondWithdrawalAndDistribute in almost same date (10-19 Nov 2021). 
3) All 29 wallets after recived DAI used HOP swapAndSend in same day (10-19 Nov 2021) with always 507-511 DAI. 

swapAndSend transactions for these 29 wallets:
0x1d9237940b102960aa4655d25f86626b9a8e743a Nov-18-2021 06:07:13 PM +UTC https://polygonscan.com/tx/0x0195d7d7236632a766957e99b39e580e00b0187ff61ad36c39f764fb6ebb80a6
0x250613f89a1543b1f97be32dc582bd3296941803 Nov-19-2021 03:51:15 AM +UTC https://polygonscan.com/tx/0x94ae3df04bc62ad37f8c04769a939dc8016e608aa140b77710aa0da2603e38a7
0x2a11d404dabbdf799bb6e0b319327f93765b5216 Nov-19-2021 03:37:31 AM +UTC https://polygonscan.com/tx/0x78693dc9fcc74abf2e872bd86657c91d94c3904ed1d5cdbcac9ab35bb70da9bd
0x2c2f46041f90062636d0edaa0b345e6f71b42933 Nov-18-2021 08:43:20 PM +UTC https://polygonscan.com/tx/0x67903334679bfd655caae35b0edc0f88495e762a2d5130d5249463de6965ec44
0x2fdaa34af555ee71ad5b87d4776e4c3e9b35368f Nov-11-2021 05:41:28 PM +UTC https://polygonscan.com/tx/0xe460cf32232ec6407b4e3afee1ec21d466aba2ead1230b5ecd49843b8ec45a4b
0x3dd52c3385c211bc5a99578d0bf898814b1e48ca Nov-18-2021 06:34:13 PM +UTC https://polygonscan.com/tx/0x17ecc3257d3d4a9dd59c2c631716499287ec202b7d7f3b879de3a3cbcf56a79f
0x55ee4f3769c870d816f2f98bf7258aa4eed8d8eb Nov-18-2021 07:38:15 PM +UTC https://polygonscan.com/tx/0xa6627610ebedfa4cae71aa1933ff5f374661bc7118479410a7071e8b4c2e8b17
0x5afffee90d48a6cd471adefb9cf45cfc3112e538 Nov-18-2021 07:57:02 PM +UTC https://polygonscan.com/tx/0xcd1cfe3a0c02b7621c9125fb3960085db5072ae5b1390c307ebffa61dfedbd3a
0x638be8bdd0ee9f614199835bd688b911a8c51793 Nov-19-2021 12:54:09 PM +UTC https://polygonscan.com/tx/0xa979a2631fae69e21c10349c888f5dda46521fd93d7f15686ab19c8c4650be14
0x6bfddb07ec564ef6b9b1f4714e3dc6390eb9faef Nov-19-2021 05:13:15 AM +UTC https://polygonscan.com/tx/0x47a374334b8866c3d9fcc0b5e301801b6da219c12f46215ebc021502b2918353
0x6c66677f31093b465f14cf1455e820e9a86501d3 Nov-10-2021 01:30:47 PM +UTC https://polygonscan.com/tx/0xf4a38e3322ac71edf7ec214930ace991a730b787d8b40eb93d89ab47e3d4caaa
0x6d45810ed704fcb7d51b60783f493e63604eb717 Nov-10-2021 12:57:19 PM +UTC https://polygonscan.com/tx/0x8684bb5ef9c73d719cf5b63c3a772a64c89139022c7afaef393fd5a47178ccfb
0x7ac1a4f9802666bfdc3ae72597e8eeee1f77b62d Nov-10-2021 02:48:23 PM +UTC https://polygonscan.com/tx/0xcd52bd5153e1ae7ce00e1e3bf1ff8cd3727f5f12b7ca07c0b2354cc3ac0dd7e2
0x7c9e6c886abb52d24b50f7e759db3af1b6459042 Nov-10-2021 06:15:32 PM +UTC https://polygonscan.com/tx/0x9846e5ed72f422a4a24b401f000978f2d41698a80153ce3739864b7495de6fbc
0x9958f1a31c70a97302370364fe701a420cfe5829 Nov-19-2021 10:59:30 AM +UTC https://polygonscan.com/tx/0xd1f112789f1e26d98d72fd54ccce7922e11fff6a0c6a1d493ab4d0f6ffa5d23e
0x9fac8fa825c16af889f0283c9bb63ba0ea437c37 Nov-10-2021 12:14:38 PM +UTC https://polygonscan.com/tx/0x06e4c67d7e3b4ab3883f42f1a2d77d381892220ca3e67ee88e24f623f9a3cd22
0xa51235b7d1087da2c6c5fbf022f48fc3458da11e Nov-10-2021 09:51:20 AM +UTC https://polygonscan.com/tx/0x7935d2ee5be6ef6cc25902b81981a1054b3c463d73f86cbcfcb1aa47aeb34744
0xa84174465090fb5dea838c45c675eaab6cc54a88 Nov-19-2021 03:14:21 AM +UTC https://polygonscan.com/tx/0x7c273cb3dd665add8b685c159f8e8f18499c1149b56c58da88af795ba4ef6068
0xab421b2e9e0eac0c0f551a6c3b088f37c19534b2 Nov-18-2021 07:18:47 PM +UTC https://polygonscan.com/tx/0x610a58adbf7320f18d0a5ae2ed85519fffff69b35881c180fd5d00a2b925dbd5
0xabcfb5692b3517449ffb192e7f7516f1bb52975e Nov-18-2021 06:45:31 PM +UTC https://polygonscan.com/tx/0x89ca8ac190bfda56b56719fa31a0b9c66c946090f6c8f4e8efce1b22065be6ed
0xae8d883df0bbb0ddea5915001eadc4c67b146d6a Nov-18-2021 07:07:47 PM +UTC https://polygonscan.com/tx/0x24a9389d2d3b660d1048e2ada42fa7977d347102b8518056f7c9aa58109ad421
0xba4ac6587413c5980ab699cf6d5dcb4562b7d6c8 Nov-11-2021 06:17:30 AM +UTC https://polygonscan.com/tx/0xed57229a990cad2d704807dd11f849a4a2c9028b9189cf3d1c63dcd5f03b80c7
0xc73765fc508528ba0b6ef13c93e53b7fc579cd81 Nov-18-2021 07:00:59 PM +UTC https://polygonscan.com/tx/0x912e2846e41f245e6a3fc219a29303225b07d2b8a5f7d79775824e97dee00db0
0xd3ba5e075b93e03b2fb0e2dc72138a37ad9efe78 Nov-18-2021 09:16:46 PM +UTC https://polygonscan.com/tx/0x18311a75389d67bbd0621b77a565effbe5d6d76bbd82c970f2e22a5e2c368108
0xd9a224313a26560bc4a737012cc68c365ad227c4 Nov-10-2021 12:34:11 PM +UTC https://polygonscan.com/tx/0xaa4b1698747249becef49294dd9f71a9637ad0531b0e8b4d8af78109919cb3d2
0xde5a9aa92c8e80a06e6cd8affa11c7695df8f337 Nov-18-2021 08:33:52 PM +UTC https://polygonscan.com/tx/0x4f9f440774c3b84b96c81e552b25bd6393d681e18818f09684ea6e078769e004
0xe3d13050bf335b9c69a9e9e77736351f73fc0e8f Nov-19-2021 04:32:31 AM +UTC https://polygonscan.com/tx/0xcecffa27715b3f4859c35d6033c996c9340713105280bbbd06e63798cd9a8866
0xece85819cbe493b488ec890c81b9583f9f1c660d Nov-11-2021 11:16:49 AM +UTC https://polygonscan.com/tx/0x188dbeaf37832848644814606415e96d08b0dc91f2e8e5ee0be9d249b531596c
0xf14ca4708db963eb5d4d78fce961c719e5703e5e Nov-10-2021 11:06:32 AM +UTC https://polygonscan.com/tx/0x574e0c395a449bc1d048e5ca34f075d6904c708f8453de5cf9ec1560346225cb

# Methodology

I checked Disperse polygon contract's transactions with polygonscan API and filtered transactions, where 20+ recipients addresses are eligible for the airdrop. Then I checked when and how this wallets interact with HOP. Node js code available on request.

# Rewards Address

0x7d3f6048091aEdE6198736b8C6ac527c314e25ff

**Comment by shanefontaine** [comment]
Thank you for your report @GreyPK. We have verified that the addresses in this report are Sybil attackers.

The report included 29 eligible addresses as Sybil attackers which means you are eligible for 3311.166639193660365902 HOP!

The qualified addresses are as follows:

```
0x1d9237940b102960aa4655d25f86626b9a8e743a
0x250613f89a1543b1f97be32dc582bd3296941803
0x2a11d404dabbdf799bb6e0b319327f93765b5216
0x2c2f46041f90062636d0edaa0b345e6f71b42933
0x2fdaa34af555ee71ad5b87d4776e4c3e9b35368f
0x3dd52c3385c211bc5a99578d0bf898814b1e48ca
0x55ee4f3769c870d816f2f98bf7258aa4eed8d8eb
0x5afffee90d48a6cd471adefb9cf45cfc3112e538
0x638be8bdd0ee9f614199835bd688b911a8c51793
0x6bfddb07ec564ef6b9b1f4714e3dc6390eb9faef
0x6c66677f31093b465f14cf1455e820e9a86501d3
0x6d45810ed704fcb7d51b60783f493e63604eb717
0x7ac1a4f9802666bfdc3ae72597e8eeee1f77b62d
0x7c9e6c886abb52d24b50f7e759db3af1b6459042
0x9958f1a31c70a97302370364fe701a420cfe5829
0x9fac8fa825c16af889f0283c9bb63ba0ea437c37
0xa51235b7d1087da2c6c5fbf022f48fc3458da11e
0xa84174465090fb5dea838c45c675eaab6cc54a88
0xab421b2e9e0eac0c0f551a6c3b088f37c19534b2
0xabcfb5692b3517449ffb192e7f7516f1bb52975e
0xae8d883df0bbb0ddea5915001eadc4c67b146d6a
0xba4ac6587413c5980ab699cf6d5dcb4562b7d6c8
0xc73765fc508528ba0b6ef13c93e53b7fc579cd81
0xd3ba5e075b93e03b2fb0e2dc72138a37ad9efe78
0xd9a224313a26560bc4a737012cc68c365ad227c4
0xde5a9aa92c8e80a06e6cd8affa11c7695df8f337
0xe3d13050bf335b9c69a9e9e77736351f73fc0e8f
0xece85819cbe493b488ec890c81b9583f9f1c660d
0xf14ca4708db963eb5d4d78fce961c719e5703e5e
```

**Comment by shanefontaine** [comment]
It looks like all addresses had similar behaviors on the Hop protocol as well:

address | ensName | total | mainnet | arbitrum | optimism | polygon | xdai | totalVolume
-- | -- | -- | -- | -- | -- | -- | -- | --
0x1d9237940b102960aa4655d25f86626b9a8e743a |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,023.93
0x3dd52c3385c211bc5a99578d0bf898814b1e48ca |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,023.92
0xba4ac6587413c5980ab699cf6d5dcb4562b7d6c8 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,022.71
0xabcfb5692b3517449ffb192e7f7516f1bb52975e |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,022.57
0xae8d883df0bbb0ddea5915001eadc4c67b146d6a |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,022.24
0x55ee4f3769c870d816f2f98bf7258aa4eed8d8eb |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,022.24
0xc73765fc508528ba0b6ef13c93e53b7fc579cd81 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,022.23
0x5afffee90d48a6cd471adefb9cf45cfc3112e538 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,021.97
0xab421b2e9e0eac0c0f551a6c3b088f37c19534b2 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,021.38
0xece85819cbe493b488ec890c81b9583f9f1c660d |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,021.37
0x2fdaa34af555ee71ad5b87d4776e4c3e9b35368f |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,021.34
0x7c9e6c886abb52d24b50f7e759db3af1b6459042 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,020.74
0x6d45810ed704fcb7d51b60783f493e63604eb717 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,020.29
0x7ac1a4f9802666bfdc3ae72597e8eeee1f77b62d |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,020.08
0x6c66677f31093b465f14cf1455e820e9a86501d3 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,020.03
0xd9a224313a26560bc4a737012cc68c365ad227c4 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,019.83
0x638be8bdd0ee9f614199835bd688b911a8c51793 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,019.06
0x2a11d404dabbdf799bb6e0b319327f93765b5216 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.83
0xd3ba5e075b93e03b2fb0e2dc72138a37ad9efe78 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.58
0x2c2f46041f90062636d0edaa0b345e6f71b42933 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.50
0xde5a9aa92c8e80a06e6cd8affa11c7695df8f337 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.50
0xa84174465090fb5dea838c45c675eaab6cc54a88 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.44
0xa51235b7d1087da2c6c5fbf022f48fc3458da11e |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.43
0x250613f89a1543b1f97be32dc582bd3296941803 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.20
0xe3d13050bf335b9c69a9e9e77736351f73fc0e8f |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.19
0x6bfddb07ec564ef6b9b1f4714e3dc6390eb9faef |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,018.13
0x9fac8fa825c16af889f0283c9bb63ba0ea437c37 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,017.81
0xf14ca4708db963eb5d4d78fce961c719e5703e5e |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,017.63
0x9958f1a31c70a97302370364fe701a420cfe5829 |   | 2 | 0 | 0 | 0 | 1 | 1 | $1,015.36

**Comment by forweb3** [comment]
sir ，One of my, our report is finding that the rule has to be a jump to survive. However, the successor has completed more than 10 user tracking records, and there is no record of him. I cancelled his reward. And this is what I found so am I not rewarded accordingly? This address has not interacted with hop. And he did not provide other information that he interacted. I report all his reports.  
0x7d3f6048091aEdE6198736b8C6ac527c314e25ff This is a brand new empty address, why can't he provide the address he interacts with, because he is the real empty hunter. He circumvented it with better technical isolation. So here's my reasoning, I want 25% of his reward to go to me and then the remaining 75% to all eligible real users. Just like our rules.
@shanefontaine
